temba/reports/views.py

```diff
@@ -0,0 +1,253 @@
+        permission = "reports.report_read"
+
+        def get_filters(self, org):
+            # get contact ids to do filter
+            values_filters = self.request.data.get("filters", {}).get("values", [])
+            groups_filters = self.request.data.get("filters", {}).get("groups", [])
+            if not any((values_filters, groups_filters)):
+                return set(), False
+
+            filter_ids = set()
+            for flow_id, fields in groupby(values_filters, lambda x: x["field"]["flow"]):
+                try:
+                    flow = org.flows.get(id=flow_id)
+                    results = flow.aggregated_results.data
+                    for field in fields:
+                        for category in field["categories"]:
+                            filter_ids.update(results.get(field["field"]["rule"], {}).get(category, []))
+                except ObjectDoesNotExist:
+                    continue
+
+            group_ids = [_id for _ids in groups_filters for _id in _ids]
+            if group_ids:
+                group_ids = org.contacts.filter(all_groups__id__in=group_ids).values_list("id", flat=True)
+                filter_ids.update(group_ids)
+            return filter_ids, True
+
+        def get_segment(self, org):
+            # get contact ids to do segment
+            segment = self.request.data.get("segment")
+            segment_ids = {}
+            if not segment:
+                return {}, False
+
+            if segment["isGroupSegment"]:
+                groups = {category["id"]: category["label"] for category in segment["categories"]}
+                groups_contacts = (
+                    org.contacts.annotate(group_id=F("all_groups__id"))
+                    .filter(group_id__in=groups.keys())
+                    .values("group_id")
+                    .annotate(contact_ids=ArrayAgg("id"))
+                )
+                for group_contacts in groups_contacts:
+                    label = groups.get(group_contacts["group_id"])
+                    if label:
+                        segment_ids[label] = set(group_contacts["contact_ids"])
+            else:
+                try:
+                    flow = org.flows.get(id=segment["field"]["flow"])
+                    results = flow.aggregated_results.data
+                    results = results.get(segment["field"]["rule"])
+                    for category in segment["categories"]:
+                        segment_ids[category["label"]] = set(results.get(category["label"], []))
+                except ObjectDoesNotExist:
+                    pass
+            return segment_ids, True
+
+        def post(self, request, *args, **kwargs):
+            user = self.request.user
+            org = user.get_org()
+
+            fields = self.request.data.get("fields", [])
+            filtered_ids, do_filter = self.get_filters(org)
+            segment_ids, do_segment = self.get_segment(org)
+            charts_data = []
```

Comment:
Even with the collection process getting complete and successful, I'm not able to see the data in some charts as you can see on the attached file below:

![Screen Shot 2021-08-12 at 19 58 01](https://user-images.githubusercontent.com/5198870/129280858-5aa48dd9-f7fa-4441-9798-207122a933a0.png)

So if it really does not have any data to show, can we add a message (like "No data") on the chart to not seem like a bug on the chart?

Reply:
Have added changes to automatically check the collecting process status and when it's completed automatically load new data. And added label instead of empty chart if there is no data to display.